easybuild/easyconfigs/p/Python/Python-3.11.5-custom-ctypes.patch

```diff
@@ -0,0 +1,122 @@
++        # haven't supplied a proper program with main(), but that does not
++        # matter.
++        expr = r'^.*%s.*$' % re.escape(name)
++        c_compiler = shutil.which('gcc')
++        if not c_compiler:
++            c_compiler = shutil.which('cc')
++        if not c_compiler:
++            # No C compiler available, give up
++            return None
++
++        temp = tempfile.NamedTemporaryFile()
++        try:
++            args = [c_compiler, '-Wl,-t', '-o', temp.name, '-l:' + name]
++            # print(args)
++            env = dict(os.environ)
++            env['LC_ALL'] = 'C'
++            env['LANG'] = 'C'
++            try:
++                proc = subprocess.Popen(args,
++                                        stdout=subprocess.PIPE,
++                                        stderr=subprocess.STDOUT,
++                                        env=env)
++            except OSError:  # E.g. bad executable
++                return None
++            with proc:
++                trace = proc.stdout.read().decode()
++        finally:
++            try:
++                temp.close()
++            except FileNotFoundError:
++                # Raised if the file was already removed, which is the normal
++                # behaviour of GCC if linking fails
++                pass
++        res = re.findall(expr, trace, re.MULTILINE)
++        if not res:
++            return None
++
++        for file in res:
++            # Check if the given file is an elf file: gcc can report
++            # some files that are linker scripts and not actual
++            # shared objects. See bpo-41976 for more details
++            if not _is_elf(file):
++                continue
++            return file
++
+     if sys.platform == "sunos5":
+         # use /usr/ccs/bin/dump on solaris
+         def _get_soname(f):
+@@ -283,7 +331,7 @@
+             abi_type = mach_map.get(machine, 'libc6')
+ 
+             # XXX assuming GLIBC's ldconfig (with option -p)
+-            regex = r'\s+(lib%s\.[^\s]+)\s+\(%s'
++            regex = r'\s+(lib%s\.[^\s]+)\s+\(%s\)\s+=>\s+(\S+)'
+             regex = os.fsencode(regex % (re.escape(name), abi_type))
+             try:
+                 with subprocess.Popen(['/cvmfs/software.eessi.io/versions/2023.06/compat/linux/x86_64/sbin/ldconfig', '-p'],
+@@ -293,7 +341,7 @@
+                                       env={'LC_ALL': 'C', 'LANG': 'C'}) as p:
+                     res = re.search(regex, p.stdout.read())
+                     if res:
+-                        return os.fsdecode(res.group(1))
++                        return os.fsdecode(res.group(2))
+             except OSError:
+                 pass
+ 
+@@ -301,7 +349,7 @@
+             # See issue #9998 for why this is needed
+             expr = r'[^\(\)\s]*lib%s\.[^\(\)\s]*' % re.escape(name)
+             cmd = ['ld', '-t']
+-            libpath = os.environ.get('LD_LIBRARY_PATH')
++            libpath = os.environ.get('LIBRARY_PATH')
```

Comment:
We get `LIBRARY_PATH` instead of `LD_LIBRARY_PATH` if `LD_LIBRARY_PATH` is filtered by the EB configuration.

Reply:
So there is a scenario where this is not what we want, and that is `libcuda.so`

Reply:
Perhaps we just add a conditional below, something like
```
             if libpath:
                 for d in libpath.split(':'):
                     cmd.extend(['-L', d]) if 'stubs' not in d.split('/')
```

Reply:
No, that doesn't help, you would still have the stubs in the list of paths and so they would be picked up over the ones coming from the OS

Reply:
The real `libcuda.so` is picked up from the default paths of the linker

Reply:
Comment added. :)